app/models/concerns/pageflow/hosted_file.rb

```diff
@@ -4,67 +4,50 @@ module HostedFile
     include UploadedFile
 
     included do
-      has_attached_file(:attachment_on_filesystem, Pageflow.config.paperclip_filesystem_default_options)
-      has_attached_file(:attachment_on_s3, Pageflow.config.paperclip_s3_default_options)
+      alias_attribute :file_name, :attachment_on_s3_file_name
 
-      validates :attachment, presence: true
+      validates :attachment_on_s3, presence: true
 
-      do_not_validate_attachment_file_type(:attachment_on_filesystem)
       do_not_validate_attachment_file_type(:attachment_on_s3)
 
-      state_machine initial: 'not_uploaded_to_s3' do
+      state_machine initial: 'uploadable' do
         extend StateMachineJob::Macro
 
-        state 'not_uploaded_to_s3'
-        state 'uploading_to_s3'
-        state 'uploaded_to_s3'
-        state 'uploading_to_s3_failed'
+        state 'uploadable'
+        state 'uploading'
+        state 'uploaded'
+        state 'uploading_failed'
 
-        event :publish do
-          transition 'not_uploaded_to_s3' => 'uploading_to_s3'
+        event :upload do
+          transition 'uploadable' => 'uploading'
         end
 
-        event :retry do
-          transition 'uploading_to_s3_failed' => 'uploading_to_s3'
+        event :publish do
+          transition 'uploading' => 'uploaded'
         end
 
-        job UploadFileToS3Job do
-          on_enter 'uploading_to_s3'
-          result :pending, retry_after: 30.seconds
-          result :ok, state: 'uploaded_to_s3'
-          result :error, state: 'uploading_to_s3_failed'
-        end
+        #job UploadFileToS3Job do
+        #  result :error, state: 'uploading_failed' # TODO: Set this in case the direct upload fails?
+        #end
 
         event :process
       end
     end
 
     def attachment
-      attachment_on_s3.present? ? attachment_on_s3 : attachment_on_filesystem
-    end
-
-    def attachment=(value)
-      self.attachment_on_filesystem = value
-    end
-
-    def keep_on_filesystem_after_upload_to_s3?
-      false
-    end
```

Comment:
This is something we will have to come back to. `Panorama::Package` [used to](https://github.com/codevise/pageflow-panorama/commit/23c5e6e7a931002f55c62dbfd25b20d82c155f15) rely on this functionality to unpack zip files. It no longer does, so this method is probably unused. Still removing it (which we pretty much have to since nothing is ever stored on the file system anymore now) will be sort of a breaking change.

Reply:
resolved with https://github.com/codevise/pageflow/pull/1082